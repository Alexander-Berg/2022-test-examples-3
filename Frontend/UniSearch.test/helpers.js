const helpers = require('../../../../hermione/utils/baobab');

async function checkNodeExists({ path, attrs }) {
    const app = await this.browser.getMeta('app');
    const isSearchApp = app === 'searchapp-phone';

    if (process.env.SPAS_AJAX && !isSearchApp) {
        path = path.replace('$page', '$subresult');
    }

    return await this.browser.yaAssertCounters(({ trees }) => {
        const check = trees.some(tree => {
            return helpers.query(
                path,
                tree.tree,
                attrs,
            ).length > 0;
        });

        if (!check) {
            const err = new Error(`В Баобаб не найден узел по пути ${path}, с аттрибутами ${JSON.stringify(attrs || {})}`);
            trees.forEach(item => helpers.getPaths(item.tree));
            err.details = { title: 'triggered counters', data: trees };
            throw err;
        }
    });
}

module.exports = {
    checkNodeExists,
};
