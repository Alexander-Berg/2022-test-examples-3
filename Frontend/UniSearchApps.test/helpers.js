const PO = require('./UniSearchApps.page-object/index');

async function openPreview(_this, serpParams, action) {
    const defaultAction = async () => _this.browser.click(PO.UniSearchApps.Content.List.Item());

    action = action || defaultAction;
    await _this.browser.yaOpenSerp(serpParams, PO.UniSearchApps());

    await _this.browser.execute(function(sel) {
        const list = document.querySelector(sel);
        const links = list.querySelectorAll('a');

        for (let i = 0; i < links.length; i++) {
            links[i].href = '#';
        }
    }, [PO.UniSearchApps.Content.List()]);

    await _this.browser.yaRetryActionsByElemVisible(
        2,
        PO.UniSearchAppsPreview(),
        action,
        3000);
}

module.exports = {
    openPreview,
};
