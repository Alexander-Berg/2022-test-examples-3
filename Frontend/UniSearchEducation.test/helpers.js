const PO = require('./UniSearchEducation.page-object/index@common');

async function openPreview(_this, serpParams, action) {
    const defaultAction = async () =>
        _this.browser.click(PO.UniSearchEducation.Content.List.Item());
    action = action || defaultAction;
    await _this.browser.yaOpenSerp(serpParams, PO.UniSearchEducation());

    await _this.browser.execute(function(sel) {
        const list = document.querySelector(sel);
        const links = list.querySelectorAll('a');

        for (let i = 0; i < links.length; i++) {
            links[i].href = '#';
        }
    }, [PO.UniSearchEducation.Content.List()]);

    await _this.browser.yaRetryActionsByElemVisible(
        2,
        PO.UniSearchEducationPreview(),
        action,
        3000);
}

module.exports = {
    openPreview,
};
