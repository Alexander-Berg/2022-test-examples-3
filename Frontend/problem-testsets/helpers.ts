import { IProblemTestset } from 'common/types/problem-test';
import { generateTestsetMapKey } from 'common/utils/helpers/problem-testsets';

export const getTestsetsList = (testsetsFullData: IProblemTestset[]) =>
    sortTestsets(testsetsFullData).map(({ id, name, sample }: IProblemTestset) => ({
        id,
        name,
        sample,
    }));

const sortTestsets = (source: IProblemTestset[]) =>
    source.sort(
        (firstTestset: IProblemTestset, secondTestset: IProblemTestset) =>
            Number(firstTestset.id) - Number(secondTestset.id),
    );

export const getTestsetMap = (testsetsFullData: IProblemTestset[], problemId: string) => {
    return testsetsFullData.reduce((accumulator, currentTestset) => {
        const { id: testsetId } = currentTestset;
        const testsetMapKey = generateTestsetMapKey(problemId, testsetId);

        return {
            ...accumulator,
            [testsetMapKey]: {
                data: currentTestset,
                fetchTestsStarted: false,
                fetchTestsError: null,
                updateTestsStarted: false,
                updateTestsError: null,
            },
        };
    }, {});
};
