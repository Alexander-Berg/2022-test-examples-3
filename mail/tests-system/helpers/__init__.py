from .fake_hub import *
