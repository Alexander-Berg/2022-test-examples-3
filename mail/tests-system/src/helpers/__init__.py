from time import sleep
import base64


def retry(tries=20, delay=0.5):
    def retry_impl(fn):
        def wrapper(*args, **kwargs):
            i = 0
            while True:
                i += 1
                try:
                    return fn(*args, **kwargs)
                except:
                    if i < tries:
                        sleep(delay)
                    else:
                        raise

        return wrapper

    return retry_impl


def decode_global_collector_id(id):
    decoded = base64.b64decode(id).decode()
    return decoded.split(":")
