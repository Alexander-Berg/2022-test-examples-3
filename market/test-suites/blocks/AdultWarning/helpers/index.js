import _ from 'lodash';
import url from 'url';
import SnippetCard from '@self/platform/spec/page-objects/snippet-card';
import ProductTopOffer from '@self/platform/spec/page-objects/n-product-top-offer';
import SnippetCard2 from '@self/project/src/components/Search/Snippet/Card/__pageObject';
import SnippetOffer from '@self/platform/components/ProductOffers/Snippet/Offer/__pageObject';

const getSnippetsCount = async ctx => {
    const cardsCount = await ctx.snippetList.getSnippetCardsLength(SnippetCard);
    const updatedCardsCount = await ctx.snippetList.getSnippetCardsLength(SnippetCard2);
    const topOffersCount = await ctx.snippetList.getSnippetCardsLength(ProductTopOffer);
    const offersCount = await ctx.snippetList.getSnippetCardsLength(SnippetOffer);

    return cardsCount + updatedCardsCount + topOffersCount + offersCount;
};

const getLastReportRequestParams = async ctx => {
    const log = await ctx.browser.getLog();
    const {Report} = log;

    const lastReportRequest = _.findLast(Report, item => {
        const {request} = item;
        const {url: requestUrl} = request;

        return requestUrl.includes('place=prime') || requestUrl.includes('place=blender');
    });

    const lastReportRequestUrl = _.get(lastReportRequest, 'request.url');
    const {query} = url.parse(lastReportRequestUrl, true);

    return query;
};

export {
    getSnippetsCount,
    getLastReportRequestParams,
};
