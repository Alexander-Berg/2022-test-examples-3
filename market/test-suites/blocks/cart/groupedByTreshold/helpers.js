// scenarios
import {buildCheckouterBucket} from '@self/root/src/spec/utils/checkouter';
import {generateString} from '@self/root/src/utils/string';

// mocks
import * as express from '@self/root/src/spec/hermione/kadavr-mock/report/express';
import * as dsbs from '@self/root/src/spec/hermione/kadavr-mock/report/dsbs';
import * as fulfilmentKettle from '@self/root/src/spec/hermione/kadavr-mock/report/kettle';
import {deliveryPostMock} from '@self/root/src/spec/hermione/kadavr-mock/checkouter/delivery';


export const FREE_DELIVERY_THRESHOLD = 2499;

export function getDsbsCart(params) {
    return getCarts({
        ...params,
        skuMock: dsbs.skuPhoneMock,
        offerMock: dsbs.offerPhoneMock,
        deliveryOptions: [{
            ...deliveryPostMock,
            isExpress: false,
            deliveryPartnerType: 'SHOP',
        }],
    });
}

export function getExpressCart(params) {
    return getCarts({
        ...params,
        skuMock: express.skuExpressMock,
        offerMock: express.offerExpressMock,
        deliveryOptions: [{
            ...deliveryPostMock,
            isExpress: true,
            deliveryPartnerType: 'YANDEX_MARKET',
        }],
    });
}

export function getFulfilmentCart(params) {
    return getCarts({
        ...params,
        skuMock: fulfilmentKettle.skuMock,
        offerMock: fulfilmentKettle.offerMock,
        deliveryOptions: [{
            ...deliveryPostMock,
            isExpress: false,
            deliveryPartnerType: 'YANDEX_MARKET',
        }],
    });
}

function getCarts({
    yaPlus = false,
    more = false,
    number = 1,
    cartIndex = 0,
    skuMock,
    offerMock,
    deliveryOptions,
}) {
    const price = getPriceByOffersNumber(more, number);
    const skuId = generateString();
    const offerId = generateString();
    const newOfferMock = {
        ...offerMock,
        wareId: offerId,
        marketSku: skuId,
        sku: skuId,
        prices: {
            ...offerMock.prices,
            value: price,
            rawValue: price,
        },
    };

    return buildCheckouterBucket({
        cartIndex,
        items: [{
            skuMock: {
                ...skuMock,
                items: [{
                    newOfferMock,
                }],
            },
            offerMock: newOfferMock,
            count: 1,
        }],
        properties: {
            yandexPlusUser: yaPlus,
        },
        deliveryOptions,
    });
}

function getPriceByOffersNumber(more, number) {
    const totalPrice = FREE_DELIVERY_THRESHOLD + (more ? (number * 10) : -(number * 10));

    return Math.floor(totalPrice / number);
}
