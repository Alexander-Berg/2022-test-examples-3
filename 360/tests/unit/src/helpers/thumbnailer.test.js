import { thumbnailer } from '../../../../src/helpers/thumbnailer';

const testFile = new File([], 'file.png', { type: 'image/png' });

const imageWidth = 1280;
const imageHeight = 1024;
const previewSize = 150;

describe('src/helpers/thumbnailer', () => {
    const originalDocumentCreateElement = document.createElement;
    const originalWindowCreateImageBitmap = window.createImageBitmap;
    const originalWindowURL = window.URL;
    const originalWindowImage = window.Image;
    const drawImage = jest.fn();
    const canvasElementMock = {
        getContext: jest.fn(() => ({
            drawImage
        })),
        toDataURL: jest.fn()
    };
    beforeEach(() => {
        window.createImageBitmap = jest.fn();
        document.createElement = () => canvasElementMock;
        window.URL = {
            createObjectURL: jest.fn()
        };
        /**
         *
         */
        function mockedImage() {
            this.width = imageWidth;
            this.height = imageHeight;
            setTimeout(() => {
                this.onload();
            });
        }
        window.Image = mockedImage;
    });
    afterEach(() => {
        jest.clearAllMocks();
        document.createElement = originalDocumentCreateElement;
        window.createImageBitmap = originalWindowCreateImageBitmap;
        window.URL = originalWindowURL;
        window.Image = originalWindowImage;
    });

    it('should call window.createImageBitmap, canvas.getContext, canvasContext.drawImage and canvas.toDataURL', async() => {
        window.createImageBitmap.mockResolvedValue({ height: imageHeight, width: imageWidth });
        await thumbnailer(testFile, previewSize);
        expect(window.createImageBitmap).toBeCalled();
        expect(canvasElementMock.getContext).toBeCalled();
        expect(drawImage).toBeCalled();
        expect(canvasElementMock.toDataURL).toBeCalled();

        expect(popFnCalls(drawImage)).toMatchSnapshot();

        expect(window.URL.createObjectURL).not.toBeCalled();
    });

    it('should use fallbackThumbnailer if window.createImageBitmap rejects', async() => {
        window.createImageBitmap.mockRejectedValue();
        await thumbnailer(testFile, previewSize);
        expect(window.createImageBitmap).toBeCalled();
        expect(window.URL.createObjectURL).toBeCalled();
        expect(canvasElementMock.getContext).toBeCalled();
        expect(drawImage).toBeCalled();
        expect(canvasElementMock.toDataURL).toBeCalled();

        expect(popFnCalls(drawImage)).toMatchSnapshot();
    });

    it('should use fallbackThumbnailer if window.createImageBitmap does not exist', async() => {
        delete window.createImageBitmap;
        await thumbnailer(testFile, previewSize);
        expect(window.URL.createObjectURL).toBeCalled();
        expect(canvasElementMock.getContext).toBeCalled();
        expect(drawImage).toBeCalled();
        expect(canvasElementMock.toDataURL).toBeCalled();

        expect(popFnCalls(drawImage)).toMatchSnapshot();
    });
});
