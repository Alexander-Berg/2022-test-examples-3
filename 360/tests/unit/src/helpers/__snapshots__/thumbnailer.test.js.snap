// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/helpers/thumbnailer should call window.createImageBitmap, canvas.getContext, canvasContext.drawImage and canvas.toDataURL 1`] = `
Array [
  Array [
    Object {
      "height": 1024,
      "width": 1280,
    },
    128,
    0,
    1024,
    1024,
    0,
    0,
    150,
    150,
  ],
]
`;

exports[`src/helpers/thumbnailer should use fallbackThumbnailer if window.createImageBitmap does not exist 1`] = `
Array [
  Array [
    mockedImage {
      "height": 1024,
      "onerror": [Function],
      "onload": [Function],
      "src": undefined,
      "width": 1280,
    },
    128,
    0,
    1024,
    1024,
    0,
    0,
    150,
    150,
  ],
]
`;

exports[`src/helpers/thumbnailer should use fallbackThumbnailer if window.createImageBitmap rejects 1`] = `
Array [
  Array [
    mockedImage {
      "height": 1024,
      "onerror": [Function],
      "onload": [Function],
      "src": undefined,
      "width": 1280,
    },
    128,
    0,
    1024,
    1024,
    0,
    0,
    150,
    150,
  ],
]
`;
